File: 04-solver-platform/api/app/schemas/net_cats.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime


class NetCatEntryIn(BaseModel):
    slot: str
    operator: str   # "Max" | "Min"
    games: int
    matchups: str   # comma-delimited "AWAY@HOME,..."


class NetCatEntryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    entry_id: str
    net_cats_id: str
    slot: str
    operator: str
    games: int
    matchups: str


class NetCatsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    net_cats_id: str
    season_id: str
    name: str
    is_snapshot: bool
    created_at: datetime
    entries: list[NetCatEntryOut]


class UpsertWorkingCopyBody(BaseModel):
    entries: list[NetCatEntryIn]


class SnapshotBody(BaseModel):
    name: str

File: 04-solver-platform/frontend/src/entities/NetCats.ts
export interface NetCatEntry {
  entry_id: string
  net_cats_id: string
  slot: string
  operator: 'Max' | 'Min'
  games: number
  matchups: string   // comma-delimited "AWAY@HOME,..."
}

export interface NetCats {
  net_cats_id: string
  season_id: string
  name: string
  is_snapshot: boolean
  created_at: string
  entries: NetCatEntry[]
}
